core/src/main/resources/org/apache/spark/ui/static/table.js

```diff
@@ -30,3 +30,75 @@ function stripeSummaryTable() {
        }
     });
 }
+
+function toggleThreadStackTrace(threadId, forceAdd) {
+    var stackTrace = $("#" + threadId + "_stacktrace")
+    if (stackTrace.length == 0) {
+        var stackTraceText = $('#' + threadId + "_td_stacktrace").html()
+        var threadCell = $("#thread_" + threadId + "_tr")
+        threadCell.after("<tr id=\"" + threadId +"_stacktrace\" class=\"accordion-body\"><td colspan=\"3\"><pre>" +
+            stackTraceText +  "</pre></td></tr>")
+    } else {
+        if (!forceAdd) {
+            stackTrace.remove()
+        }
+    }
+}
+
+// expandOrCollapse - true: expand, false: collapse
+function expandOrCollapseAllThreadStackTrace(expandOrCollapse, toggleButton) {
```

Comment:
so I am totally unqualified to be judging javascript code, so feel free to reject this suggestion -- but to me it kinda seems like this really two different functions based on `expandOrCollapse`.  The callsite always statically knows the value of that param, so it doesn't help to put them together.  Might be cleaner to separate them out?  Also it seems like its never called with `true, false`, so then you could eliminate that option.

(But again, totally up to your judgement, I have an incomplete understanding of what is going on here ...)